lib/pact/consumer_contract/request_decorator.rb

```diff
@@ -20,25 +20,35 @@ def as_json options = {}
       hash[:query]   = query   if request.specified?(:query)
       hash[:headers] = headers if request.specified?(:headers)
       hash[:body]    = body    if request.specified?(:body)
-      hash
+      include_matching_rules? ? with_matching_rules(hash) : hash
     end
 
     private
 
     attr_reader :request
 
     def path
```

Comment:
I don't think these if statements are required. Doing a reification on an object that is not a term will just return that object. I didn't have these lines in my example - did you find a use case that required them?